src/parse/parse.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+#include <cstdio>
+#include <type_traits>
+#include <memory>
+#include "core/routine.hpp"
+#include "core/cache.hpp"
+
+namespace docgen {
+namespace parse {
+namespace details {
+
+struct file_deleter
+{
+    void operator()(FILE* file) const
+    {
+        fclose(file);
+    }
+};
+
+} // namespace details
+
+// Parse a file given a filepath.
+void parse_file(const char* filepath)
+{
+    constexpr size_t buf_size = 1000;   // TODO: experiment later optimal size
+    constexpr size_t symbol_size = 10;  // TODO: figure out optimal symbol size statically later
+
+    using Routine = core::Routine;
+    using cache_t = core::Cache<symbol_size>;
+    using routine_array_t = core::routine_array_t<cache_t>;
+
+    static constexpr routine_array_t routines = core::make_routines<cache_t>();
+
+    cache_t cache;
+    Routine routine = Routine::READ;    // initially in read routine
+
+    using file_ptr_t = std::unique_ptr<FILE, details::file_deleter>;
```

Comment:
No.

Reply:
We only use `fopen()` and `fclose()` once in this entire program. Programming in a language does not mean religiously obeying common practices—you use the language as a tool, as it pertains to what you're doing. Destructors are useful when you will be dealing with a certain object a lot, in different scopes, to the point where it can get confusing and error prone—this simply isn't the case here. The entire lifetime of this FILE pointer exists within this one function, and we never use them anywhere else. I'm not going down this road, where we are adding unnecessary/convoluted code, with no tangible benefit, simply because you think Bjarne would like it. What you've done in this instance is turn one line into a dozen—nothing more. It's just silly.

Reply:
How is this convoluted... 

But in any case, I get what you're saying about religiously obeying common practices. We're spending too much time on this, I'll just use `fclose`.